pkg/dogstatsd/parser.go

```diff
@@ -278,7 +277,7 @@ func parseMetricMessage(message []byte) (*metrics.MetricSample, error) {
 		}
 	}
 
-	metricName := string(rawName)
+	metricName := namespace + string(rawName)
```

Comment:
This is the slowest way to build strings on go - we should probably rely on a stringbuffer given the frequency at which this function would be called (once for each dogstatsd message). We should also probably not concatenate at all if the `namespace==""` - reference: https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go

Reply:
Since you guys are out, I'll implement the fix @ChristineTChen :)